tests/unit/s2n_ktls_test_utils_test.c

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "utils/s2n_random.h"
+
+#define S2N_TEST_RECORD_TYPE 43
+
+#define INIT_MSGHDR(name, buf, len)       \
```

Comment:
This macro makes it really difficult to tell what is going on and you're only using it three times. I think you're trying not to repeat code in between tests, but if you're sharing stuff just initialize it before the test.

Reply:
I would like to create a new msghdr so that each one is a clean slate. This macro gets used alot in the next PR also and its not super interesting (create msghdr, iovec and set some values).

Is it difficult because its a new pattern or something else? Anything I can do to make it nicer? Comment maybe?

Reply:
Here is the code without the macro. Lmk what you think because I still have a preference for the macro and I think we would want it in the next PR as we have alot more tests and alot more of this verbose init code.

https://github.com/aws/s2n-tls/pull/4109/commits/3aa407786791c1d9aef325215a38c042082b1876

We could reach a compromise with a function but that still atlease 3 lines of code and more for the function so prob doesnt help too much. I do wonder if the macro is glaring because its different and maybe its less difficult if we become normalized to it.